### ISSUE TITLE
JavaFX  was removed in JDK 11

### ISSUE THREAD

**Issue by miurahr** [yoursdearboy/omegat-browser#14]
Oracle JDK and OpenJDK 11 removed JavaFX, and Java 9-10 is deprecated now.

>Note: Java 8 or newer required with JavaFX support. Oracle's Java 8 to 10 already have JavaFX. For other distributions or newer versions download it from Gluon. On Linux it is avaiable via openjfx package.
More info can be found in this thread.

Major Linux distributions and Java IDE, such as IDEA, have already used its default jdk to version  11.
So a launch of the plugin cause NoClassDefFoundError.

```
11816: Error: java.lang.NoClassDefFoundError: javafx/application/Platform 
11816: Error: 	at com.yoursdearboy.omegat.plugins.browser.BrowserPlugin$2.onApplicationStartup(BrowserPlugin.java:40) 
```

**Comment by yoursdearboy** [comment]
Hi @miurahr. I'm glad to see your PR, it's awesome that somebody fixed this JavaFX hell. Could you combine both your PR's in one so I can safely merge it?

**Comment by miurahr** [comment]
OK, I'd like to consolidate two PR to #15.

**Comment by SergioDaroca** [comment]
java --module-path /usr/share/openjfx/lib --add-modules=javafx.base,javafx.controls,javafx.graphics,javafx.fxml,javafx.swing,javafx.web,javafx.media --add-exports=javafx.graphics/com.sun.glass.events=ALL-UNNAMED -jar OmegaT.jar

worked for me for for OmegaT_5.4.1_Beta_Linux_64 on Ubuntu 20.10

/usr/share/openjfx/lib was the javafx path after installing it with

sudo apt-get install openjfx

I needed --add-exports=javafx.graphics/com.sun.glass.events=ALL-UNNAMED
to address this error, so it could recognize Key Events to open websites:

 [...] module javafx.graphics does not export com.sun.glass.events to unnamed module [...]

This article by Simon Ritter helped:
https://medium.com/azulsystems/using-jlink-to-build-java-runtimes-for-non-modular-applications-9568c5e70ef4

------------------------------ NORWEGIAN SCRIPTS -----------------------------------

Added a couple scripts for norwegian websites naob.no and ordbok.uib.no:
[naob.no.groovy.txt](https://github.com/yoursdearboy/omegat-browser/files/5736667/naob.no.groovy.txt)
[ordbok.uib.no.groovy.txt](https://github.com/yoursdearboy/omegat-browser/files/5736670/ordbok.uib.no.groovy.txt)

**Comment by SergioDaroca** [comment]
It's a wonderful plugin for a wonderful project! I'll be contributing as soon as I can!

**Comment by miurahr** [comment]
#15 make the project generate fat-jar with javaFx binaries that works with JRE 11 and later.
@SergioDaroca  mentions that you can configure java to run with openjfx on your platform with a *released* omegat-browser.
Does these two conflict each other?

**Comment by SergioDaroca** [comment]
Hi @miurahr !
I used the plugin version 1.3 with OmegaT 4.3 in an older computer, and it went perfect.
Then I moved to a newer computer with Ubuntu and Java 11, and the menus didn't appear and I had some error messages, like the one mentioned up here:

Error: java.lang.NoClassDefFoundError: javafx/application/Platform 

I got it all working after investigating with the recipe I wrote up here at:
https://github.com/yoursdearboy/omegat-browser/issues/14#issuecomment-750427448

, so actually I don't know if the 1.3 plugin has compiled javafx? Or if you mean it comes with OmegaT itself, I don't know! My OmegaT install does have a JRE folder but I don't find any javafx related files there.
What I wrote up here worked for me.
And I hope it helps others! 'Cause it's a great software.

Also, I made a custom .desktop file to add a shortcut to the Ubuntu menu
with the help of this page:
https://averagelinuxuser.com/ubuntu_custom_launcher_dock/

So in

sudo nano /usr/share/applications/omegat.desktop

I pasted this:

#!/usr/bin/env xdg-open
[Desktop Entry]
Version=1.0
Type=Application
Terminal=false
Exec=java --module-path /usr/share/openjfx/lib --add-modules=javafx.base,javafx.controls,javafx.graphics,javafx.fxml,javafx.swing,javafx.web,javafx.media --add-exports=javafx.graphics/com.sun.glass.events=ALL-UNNAMED -jar /home/sergio/Programas/OmegaT_5.4.1_Beta_Linux_64/OmegaT.jar 
Name=OmegaT
Comment=Best Computer Aided Translation
Icon=/home/sergio/Programas/OmegaT_5.4.1_Beta_Linux_64/images/OmegaT.iconset/icon_128x128.png

then
sudo chmod +x /usr/share/applications/omegat.desktop
then run it from the Ubuntu dash menu and then "Add to Favorites” and pinned it to the dock panel

Took me some time and effort, but I'm translating really comfortably now. ;)

**Comment by miurahr** [comment]
@SergioDaroca could you build version 1.4-SNAPSHOT on master branch of github repository from source to get #15 improvment?

**Comment by SergioDaroca** [comment]
@miurahr I'm without a good connection and time for a couple weeks, but when i'm back on the real world, what IDE etc... are you using to build the plugin, I've never been too much into java / groovy, but I've used eclipse before, and I'm on Ubuntu these days...

**Comment by SergioDaroca** [comment]
@miurahr @yoursdearboy 
Hei! I'm installing Eclipse and will try to compile the 1.4-SNAPSHOT, wish me luck!

**Comment by SergioDaroca** [comment]
So, forget using eclipse, I downloaded the project, ran ./gradlew, which downloaded gradle, then ran ./gradlew build, which created a .zip and a .tgz with the omegat-browser-1.4.0-rc.1.jar inside. Did the same fot rc.2
Placed omegat-browser-1.4.0-rc.1.jar at ~/.omegat/plugins
I'm running the **OmegaT_5.4.3_Beta_Without_JRE**  in Ubuntu with last updates april 2021, which broke my former configuration, and using this java:
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+9-Ubuntu-0ubuntu1.20.10)
OpenJDK 64-Bit Server VM (build 11.0.10+9-Ubuntu-0ubuntu1.20.10, mixed mode, sharing)

Modified /usr/share/applications/omegat.desktop:

#!/usr/bin/env xdg-open
[Desktop Entry]
Version=5.4.3
Type=Application
Terminal=false
Exec=java --module-path /usr/share/openjfx/lib --add-modules=javafx.base,javafx.controls,javafx.graphics,javafx.fxml,javafx.swing,javafx.web,javafx.media --add-exports=javafx.graphics/com.sun.glass.events=ALL-UNNAMED -jar /home/sergio/Programas/OmegaT_5.4.3_Beta_Without_JRE/OmegaT.jar
Name=OmegaT
Comment=Best Computer Aided Translation
Icon=/home/sergio/Programas/OmegaT_5.4.3_Beta_Without_JRE/images/OmegaT.iconset/icon_128x128.png

Now it works fine, using the modified .desktop file. All my browser scripts work, but when I enable the Google Translator tab the app freezes. I wouldn't use it anyways, just wondering why... probably google changed something in their end?
Keybinding in scripts work fine too.

The command used in the .desktop file is:

$ java --module-path /usr/share/openjfx/lib --add-modules=javafx.base,javafx.controls,javafx.graphics,javafx.fxml,javafx.swing,javafx.web,javafx.media --add-exports=javafx.graphics/com.sun.glass.events=ALL-UNNAMED -jar /home/sergio/Programas/OmegaT_5.4.3_Beta_Without_JRE/OmegaT.jar

There's some terminal warnings and info:

63895: Info: =================================================================== 
63895: Info: OmegaT-5.4.3_0_dc1f7028 (Mon Apr 19 17:59:40 CEST 2021)  Locale en_US 
63895: Info: Java: Ubuntu ver. 11.0.10, executed from '/usr/lib/jvm/java-11-openjdk-amd64' (LOG_STARTUP_INFO)
63895: Info: Jar used for plugins: file:/home/sergio/.omegat/plugins/omegat-browser-1.4.0-rc.1.jar (PLUGIN_LOAD_JAR)


63895: Info: Docking Framework version: 2.1.5 
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.omegat.Main (file:/home/sergio/Programas/OmegaT_5.4.3_Beta_Without_JRE/OmegaT.jar) to field sun.awt.X11.XToolkit.awtAppClassName
WARNING: Please consider reporting this to the maintainers of org.omegat.Main
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Gtk-Message: 17:59:41.803: Failed to load module "canberra-gtk-module"


63895: Info: Click on 'toolsMenu' menu item (LOG_MENU_CLICK)
63895: Info: Enabling browser script: google_translate.groovy 
63895: Info: New domain consent.google.com doesn't match translate.google.com. Redirecting back. xdxv

As said, when enabling the Google translate browser script the program freezes and won't close, I have to close it using the System Monitor process list.

Again, great plugin for a great program.

**Comment by SergioDaroca** [comment]
Forgot to say, it works the same with rc.2

**Comment by SergioDaroca** [comment]
Also, running rc.2 and rc.1 with the OmegaT Java included release 5.4.3 Beta had some issues.
Window was minimum size, again browser scripts are not in menu, here's feedback:

========================================================
19416: Info: OmegaT-5.4.3_0_dc1f7028 (Tue Apr 20 11:50:03 CEST 2021)  Locale en_US 
19416: Info: Java: AdoptOpenJDK ver. 1.8.0_282, executed from '/home/sergio/Programas/OmegaT_5.4.3_Beta_Linux_64/jre' (LOG_STARTUP_INFO)
19416: Info: Jar used for plugins: file:/home/sergio/.omegat/plugins/omegat-browser-1.4.0-rc.2.jar (PLUGIN_LOAD_JAR)


19416: Error: java.lang.UnsupportedClassVersionError: javafx/application/Platform has been compiled by a more recent version of the Java Runtime (class file version 54.0), this version of the Java Runtime only recognizes class file versions up to 52.0 


Guess it's fixed with the code in this comment?  https://github.com/yoursdearboy/omegat-browser/issues/3#issuecomment-792150445
